File: src/141009_src/FizzBuzz/fizzBuzz.js
var fizzBuzz = function(n) {
  if (n % 15 === 0) {
    return "FizzBuzz";
  } else if (n % 5 === 0) {
    return "Buzz";
  } else if (n % 3 === 0) {
    return "Fizz";
  } else {
    return n
  }
};

// var fizzBuzz = function(n) {
//   return n % 3 === 0 ? (n % 5 === 0 ? "FizzBuzz" : "Fizz") : (n % 5 === 0 ? "Buzz" : n);
// };
//

for (var i = 1; i <= 100; i++) {
  console.log(fizzBuzz(i));
}

File: src/141009_src/FizzBuzz/fizzBuzz.scala
/**
 * scalaで書いたfizzbuzzのサンプル
 */
object FizzBuzz extends App {
  def fizzbuzz(n: Int) : String = {
    if (n % 15 == 0) {
      "FizzBuzz"
    } else if (n % 5 == 0) {
      "Buzz"
    } else if (n % 3 == 0) {
      "Fizz"
    } else {
      n.toString()
    }
  }

  // val fizzbuzz = (n:Int) => {
  //   if (n % 15 == 0) {
  //     "FizzBuzz"
  //   } else if (n % 5 == 0) {
  //     "Buzz"
  //   } else if (n % 3 == 0) {
  //     "Fizz"
  //   } else {
  //     n.toString()
  //   }
  // }:String

  (1 to 100).foreach { i =>
    println(fizzbuzz(i))
  }

}
